File: Programmers/Lv.1/크레인 인형뽑기 게임/64061.java
import java.util.*;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        
        List<Integer> list = new ArrayList<>();
        
        for (int m : moves) {
            for (int i = 0; i < board.length; i++) {
                if (board[i][m - 1] != 0) {
                    list.add(board[i][m - 1]);
                    board[i][m - 1] = 0;
                    
                    int size = list.size();
                    if (size >= 2 && list.get(size - 1) == list.get(size - 2)) {
                        answer += 2;
                        
                        list.remove(size - 1);
                        list.remove(size - 2);
                    }
                    
                    break;
                }
            }
        }
        
        return answer;
    }
}

File: Programmers/Lv.1/크레인 인형뽑기 게임/64061.js
function solution(board, moves) {
    let answer = 0;
    
    const list = [];
    
    moves.forEach(m => {
       for (let i = 0; i < board.length; i++) {
           if (board[i][m - 1] !== 0) {
               list.push(board[i][m - 1]);
               board[i][m - 1] = 0;
               
               const l = list.length;
               if (l >= 2 && list[l - 1] === list[l - 2]) {
                   answer += 2;
                   
                   list.pop(l - 1);
                   list.pop(l - 2);
               }
               
               break;
           }
       } 
    });
    
    return answer;
}
